spec/javascript/components/supplemental_application/sections/RentalAssistance.test.js

```diff
@@ -0,0 +1,78 @@
+/* global wait */
+/* global mount */
+import React from 'react'
+import renderer from 'react-test-renderer'
+import Context from '~/components/supplemental_application/context'
+import RentalAssistance from '~/components/supplemental_application/sections/RentalAssistance'
+import { cloneDeep } from 'lodash'
+
+const baseContext = {
+  rentalAssistances: [],
+  applicationMembers: [],
+  addNewRentalAssistance: false,
+  handleOpenRentalAssistancePanel: () => { },
+  handleCloseRentalAssistancePanel: () => { },
+  handleSaveNewRentalAssistance: () => { },
+  showAddRentalAssistanceBtn: () => { },
+  hideAddRentalAssistanceBtn: () => { }
+}
+
+describe('RentalAssistance', () => {
+  test('should not render a table if rental assistances is empty', () => {
+    const context = cloneDeep(baseContext)
+    const component = renderer.create(
+      <Context.Provider value={context}>
+        <RentalAssistance />
+      </Context.Provider>
+    )
+
+    let tree = component.toJSON()
+    expect(tree).toMatchSnapshot()
```

Comment:
how do we check that this works as expected? the snapshot has a lot of lines in it 😕

Reply:
i agree the snapshots do contain a fair bit of info, and i think that's sort of the point - if we want to test the exact rendered state of a component, we'll have to check its exact rendered state. when i review snapshots that are newly generated or updated, i open the page containing the component in question, and review that side by side with the snapshot lines, to check that the snapshot lines match what I see on the page, and to check that what I see on the page also matches what is intended in the pivotal story. it forces me to understand very precisely what the correct, intended render is for that component. i'm definitely open to exploring new ways to test the exact render state of components too! or to exploring other kinds of tests that could somehow fulfill or obviate the need to test the rendering of our components. do you have any thoughts around that?

Reply:
this seems like a good solution for now, i'll work on being more careful with my review of new snapshots